### PR TITLE
Update gravity.sh

### PR DESCRIPTION
**Pull request by jpgpi250**
Signed-off-by: jpgpi250 <jpgpi250@gmail.com>

**By submitting this pull request, I confirm the following:** 
*please fill any appropriate checkboxes, e.g: [X]*

- [x ] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md), as well as this entire template.
- [x ] I have made only one major change in my proposed changes.
- [x ] I have commented my proposed changes within the code.
- [x ] I have tested my proposed changes, and have included unit tests where possible.
- [x ] I am willing to help maintain this change if there are issues with it later.
- [x ] I give this submission freely and claim no ownership.
- [x ] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [x ] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

Please make sure you [Sign Off](https://docs.pi-hole.net/guides/github/how-to-signoff/) all commits. Pi-hole enforces the [DCO](https://docs.pi-hole.net/guides/github/contributing/).

---
**What does this PR aim to accomplish?:**
remove trailing periods from domain entries, if an adlist contains such entries, see issue https://github.com/pi-hole/pi-hole/issues/4701
**How does this PR accomplish the above?:**
the required change has been explained in the github issue, summary, change the sed command (line 531) in gravity.sh

**What documentation changes (if any) are needed to support this PR?:**
none

---
* You must follow the template instructions. Failure to do so will result in your pull request being closed.
* Please respect that Pi-hole is developed by volunteers, who can only reply in their spare time.

### CODE REVIEW

File: gravity.sh [hunk 1]
```diff
@@ -528,7 +528,9 @@ parseList() {
   # 1. Remove all domains containing invalid characters. Valid are: a-z, A-Z, 0-9, dot (.), minus (-), underscore (_)
   # 2. Append ,adlistID to every line
   # 3. Ensures there is a newline on the last line
```
Comment (yubiuser) [inline gravity.sh:530]:
```suggestion
  # 3. Remove trailing period (see https://github.com/pi-hole/pi-hole/issues/4701)
  # 4. Ensures there is a newline on the last line
```

File: gravity.sh [hunk 2]
```diff
@@ -528,7 +528,9 @@ parseList() {
...
-  sed -e "/[^a-zA-Z0-9.\_-]/d;s/$/,${adlistID}/;/.$/a\\" "${src}" >> "${target}"
+  # sed -e "/[^a-zA-Z0-9.\_-]/d;s/$/,${adlistID}/;/.$/a\\" "${src}" >> "${target}"
```
Comment (yubiuser) [inline gravity.sh:531]:
```suggestion
```

File: gravity.sh [hunk 3]
```diff
@@ -528,7 +528,9 @@ parseList() {
...
+  # previous line modified 20/04/2022, see https://github.com/pi-hole/pi-hole/issues/4701
```
Comment (yubiuser) [inline gravity.sh:532]:
```suggestion
```

**Review by yubiuser — CHANGES_REQUESTED** [review]

**Comment by jpgpi250** [comment]
I have applied all of your changes, unfortunately, DCO now fails. Now idea how I should fix this...

**Comment by yubiuser** [comment]
Your first commit was signed. It failed because of the commits directly on github (reason?). I manually set it to pass.

**Comment by jpgpi250** [comment]
Thanks to all for your valuble input and help with this.

**Review by yubiuser — APPROVED** [review]

**Pull request merged**